### PR TITLE
Fix misuses of a vs an

### PR DESCRIPTION
**Pull request by cuishuang**
Fixes the misuse of "a" vs "an", according to English grammatical
expectations and using https://www.a-or-an.com/

### CODE REVIEW

File: compiler/rustc_codegen_gcc/src/back/write.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ pub(crate) unsafe fn codegen(cgcx: &CodegenContext<GccCodegenBackend>, _diag_han
                     println!("Dumping reproducer {}", module.name);
                     let _ = fs::create_dir("/tmp/reproducers");
                     // FIXME(antoyo): segfault in dump_reproducer_to_file() might be caused by
-                    // transmuting an rvalue to an lvalue.
+                    // transmuting a rvalue to a lvalue.
```
Comment (GuillaumeGomez) [inline compiler/rustc_codegen_gcc/src/back/write.rs:52]: I think you got it wrong here. At least in gcc docs it's both `an`. For example: https://gcc.gnu.org/onlinedocs/gcc-3.3.6/gcc/Lvalues.html

File: compiler/rustc_ast_pretty/src/pp.rs [hunk 2]
```diff
@@ -90,7 +90,7 @@
 //!
 //! That is, in the consistent-break blocks we value vertical alignment
 //! more than the ability to cram stuff onto a line. But in all cases if it
-//! can make a block a one-liner, it'll do so.
+//! can make a block an one-liner, it'll do so.
```
Comment (kadiwa4) [inline compiler/rustc_ast_pretty/src/pp.rs:93]: that's not right

File: compiler/rustc_builtin_macros/src/source_util.rs [hunk 3]
```diff
@@ -188,7 +188,7 @@ pub fn expand_include_str(
                 base::MacEager::expr(cx.expr_str(sp, interned_src))
             }
             Err(_) => {
-                cx.span_err(sp, format!("{} wasn't a utf-8 file", file.display()));
+                cx.span_err(sp, format!("{} wasn't an utf-8 file", file.display()));
```
Comment (kadiwa4) [inline compiler/rustc_builtin_macros/src/source_util.rs:191]: that's not right; the U in UTF is pronounced like you

File: compiler/rustc_codegen_llvm/src/back/write.rs [hunk 4]
```diff
@@ -670,7 +670,7 @@ pub(crate) unsafe fn codegen(
         // A codegen-specific pass manager is used to generate object
         // files for an LLVM module.
         //
-        // Apparently each of these pass managers is a one-shot kind of
+        // Apparently each of these pass managers is an one-shot kind of
```
Comment (kadiwa4) [inline compiler/rustc_codegen_llvm/src/back/write.rs:673]: that's not right

File: compiler/rustc_codegen_llvm/src/builder.rs [hunk 5]
```diff
@@ -31,7 +31,7 @@ use std::iter;
 use std::ops::Deref;
 use std::ptr;
 
-// All Builders must have an llfn associated with them
+// All Builders must have a llfn associated with them
```
Comment (kadiwa4) [inline compiler/rustc_codegen_llvm/src/builder.rs:34]: I think it's usually pronounced ell-ell, which would mean "an" makes more sense here

**Review by kadiwa4 — CHANGES_REQUESTED** [review]

File: compiler/rustc_codegen_gcc/src/back/write.rs (hunk 1 above)
Comment (bjorn3) [reply compiler/rustc_codegen_gcc/src/back/write.rs:52]: AFAIK a vs an is based on pronounciation, not how it is written. rvalue is pronounced as ar-value and lvalue as el-value, so they get an "an", not an "a" as you would expect from how it is written.

Disclaimer: not a native english speaker.

File: compiler/rustc_codegen_llvm/src/back/write.rs (hunk 4 above)
Comment (cuishuang) [reply compiler/rustc_codegen_llvm/src/back/write.rs]: Thanks. 

Fixed

File: compiler/rustc_builtin_macros/src/source_util.rs (hunk 3 above)
Comment (cuishuang) [reply compiler/rustc_builtin_macros/src/source_util.rs]: Thank you for your patient guidance, it has been fixed

File: compiler/rustc_ast_pretty/src/pp.rs (hunk 2 above)
Comment (cuishuang) [reply compiler/rustc_ast_pretty/src/pp.rs]: Thnaks, Done

File: compiler/rustc_codegen_gcc/src/back/write.rs (hunk 1 above)
Comment (cuishuang) [reply compiler/rustc_codegen_gcc/src/back/write.rs]: Thank you for your patient guidance, it has been fixed

Comment (cjgillot) [reply compiler/rustc_codegen_gcc/src/back/write.rs]: This comment applies to "an rvalue" too.

Comment (cuishuang) [reply compiler/rustc_codegen_gcc/src/back/write.rs]: Thanks for the guidance, modified~

File: config.example.toml [hunk 6]
```diff
@@ -730,7 +730,7 @@ changelog-seen = 2
 #llvm-config = <none> (path)
 
 # Override detection of whether this is a Rust-patched LLVM. This would be used
-# in conjunction with either an llvm-config or build.submodules = false.
+# in conjunction with either a llvm-config or build.submodules = false.
```
Comment (euclio) [inline config.example.toml:733]: Same here.

Comment (cuishuang) [reply config.example.toml]: Modified.   Thanks very much.

File: compiler/rustc_codegen_llvm/src/coverageinfo/mod.rs [hunk 7]
```diff
@@ -245,7 +245,7 @@ fn add_unused_function_coverage<'tcx>(
 }
 
 /// Calls llvm::createPGOFuncNameVar() with the given function instance's
-/// mangled function name. The LLVM API returns an llvm::GlobalVariable
+/// mangled function name. The LLVM API returns a llvm::GlobalVariable
```
[the remark this answers is not in the document]
Comment (cuishuang) [reply compiler/rustc_codegen_llvm/src/coverageinfo/mod.rs]: Modified.   Thanks very much.

**Pull request merged**